### DISCUSSION TITLE
How To Use Mapster mapper in Gridify

### DISCUSSION THREAD

**Discussion by Mr-Mohammad** [alirezanet/Gridify#110] [Q&A] [answered]
hi @alirezanet 
How can I Use Mapster mapper with a Gridify?
for example:

```
   public async Task<Paging<ApplicantDocumentDto>> GetCartable(GridifyQuery query)
        {
            var result= userCartableRepository.GetCartable(query);

        }
```

**How To Map resalt to Paging<ApplicantDocumentDto>**

**Comment by alirezanet** [comment]
Hi @Mr-Mohammad,

it is similar to [AutoMapper example](https://alirezanet.github.io/Gridify/guide/autoMapper.html#gridifyto), 

e.g
```csharp
public static Paging<TDestination> GridifyTo<TSource, TDestination>(this IQueryable<TSource> query,
    TypeAdapterConfig config, IGridifyQuery gridifyQuery, IGridifyMapper<TSource> mapper = null)
{
    var res = query.GridifyQueryable(gridifyQuery, mapper);

    // Map the query results to the destination type using Mapster
    var mappedResults = res.Query.ProjectToType<TDestination>(config);

    return new Paging<TDestination>(res.Count, mappedResults);
}
```

**Comment by Mr-Mohammad** [comment]
Recursive type result is
` Paging<ApplicantDocument>
`I want Map to
` Paging<ApplicantDocumentDto>`
 
How To use GridifyTo Without  TypeAdapterConfig ??

**Reply by alirezanet** [reply]
It depends on how you are using Mapster in your project, 
let's assume you don't want to use Gridify, how do you normally do this conversion? (e.g give me an example where you returning ApplicantDocumentDto[] ...)

By the way GridifyTo is just a helper extension method, GridifyQueryable() method is important since it gives you the `count` and the query separately. you can use the query with any object mapper. then you can **manually create the paging** like above example

**Reply by alirezanet** [reply, marked as answer]
```csharp
// execute the query
 var res = query.GridifyQueryable(gridifyQuery, mapper);

 var mappedResults = ....  // Map res.Query with any mapper you want

// create the paging
    return new Paging<TDestination>(res.Count, mappedResults);
```
